### PR TITLE
Initial trace implementation

### PR DESCRIPTION
**Pull request by h2non**
Take a look to the HTTP fields that are registered.
Also, not use if should use JSON as default formatter.  The trace output is be to `stdout`.

Trace log level is `info`. Perhaps we want to change or customize this.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,8 @@
 # trace [![Build Status](https://travis-ci.org/vinxi/trace.png)](https://travis-ci.org/vinxi/trace) [![GoDoc](https://godoc.org/github.com/vinxi/trace?status.svg)](https://godoc.org/github.com/vinxi/trace) [![Coverage Status](https://coveralls.io/repos/github/vinxi/trace/badge.svg?branch=master)](https://coveralls.io/github/vinxi/trace?branch=master) [![Go Report Card](https://goreportcard.com/badge/github.com/vinxi/trace)](https://goreportcard.com/report/github.com/vinxi/trace)
 
-HTTP traffic tracing instrumnetation for your proxies. 
+Traffic tracing instrumentation for your proxies.  Designed to be extended to trace custom data or modify the request/response.
```
Comment (groyoh) [inline README.md]: Double space

File: trace.go [hunk 2]
```diff
@@ -0,0 +1,124 @@
+// Package trace implements a featured and structured generic logger
```
Comment (groyoh) [inline trace.go]: Leftover from log?

File: trace.go [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+		}
+	}
+
+	// Continue with next middleware
+	h.ServeHTTP(w, r)
+}
+
+// DefaultTracer reads the request info and writes a structured data entry to the default logger.
+// It's used as default tracer function. Returns an entry logger for subsequent writes.
+func DefaultTracer(l log.Interface, w http.ResponseWriter, r *http.Request) log.Interface {
```
Comment (groyoh) [inline trace.go]: Does it make sense that this is exported?

**Comment by groyoh** [comment]
Looks good to me! :+1:

File: trace.go (hunk 3 above)
Comment (h2non) [inline trace.go]: In this case it doesn't. Good catch.

**Pull request merged**